src/Notmuch/Binding.chs

```diff
@@ -356,10 +358,15 @@ thread_get_messages ptr = liftIO $ withThread ptr $ \ptr' ->
     >>= messagesToList . Messages
 
 -- notmuch_thread_get_matched_messages -> Int
--- notmuch_thread_get_authors -> String
--- notmuch_thread_get_subject
+
+thread_get_authors :: Thread a -> IO B.ByteString
+thread_get_authors ptr = withThread ptr ({#call thread_get_authors #} >=> B.packCString)
+
+thread_get_subject :: Thread a -> IO B.ByteString
+thread_get_subject ptr = withThread ptr ({#call thread_get_subject #} >=> B.packCString)
```

Comment:
you probably need to `detachPtr` the result of `thread_get_authors` and `thread_get_subject`.  This is likely the cause of the segfaults.

Reply:
Cheers @frasertweedale for the suggestion.

I've tried to figure out what `detachPtr` does but have trouble. Is it basically a cast from a given pointer type to another?

Also, tried your suggestion and unless I've stuffed it up run into the same problem:
```Haskell
thread_get_authors :: Thread a -> IO B.ByteString
thread_get_authors ptr = withThread ptr $ \ptr' ->
  ({#call thread_get_authors #} ptr'
   >>= detachPtr
   >>= B.packCString)
```